### PR TITLE
HPCC-25519 Build nightly master docker image

### PR DESCRIPTION
**Pull request by jakesmith**
Signed-off-by: Jake Smith <jake.smith@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:

Have run this action (with minor mods to target my docker repo) on my github repo. for a few days.

### CODE REVIEW

**Comment by jakesmith** [comment]
@richardkchapman - please review.

**Comment by jakesmith** [comment]
dfse2.ecl caused the regression suite to timeout:
```
2021-03-15T12:36:36.0577381Z [Action] 164. Test: dfse2.ecl
...
2021-03-15T12:48:36.0807129Z [Failure] 164. Create Stack Trace for thormaster_mythor (pid:10117) into '/home/runner/work/HPCC-Platform/HPCC-Platform/regress/log/W20210315-123636-thormaster_mythor.trace'
2021-03-15T12:48:39.0102017Z [Failure] 164. Create Stack Trace for thorslave_mythor1 (pid:10105) into '/home/runner/work/HPCC-Platform/HPCC-Platform/regress/log/W20210315-123636-thorslave_mythor1.trace'
2021-03-15T12:48:41.7065077Z [Failure] 164. Create Stack Trace for thorslave_mythor2 (pid:10110) into '/home/runner/work/HPCC-Platform/HPCC-Platform/regress/log/W20210315-123636-thorslave_mythor2.trace'
2021-03-15T12:48:44.3369607Z [Failure] 164. Timeout occured for dfse2.ecl and no more attempt left. Force to abort... 
```

But it failed immediately (as designed) with:
```
Error (1001): Command return code non-zero. 
 retCode: 1
'eclcc: /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/ecl/dfse2.ecl(32,12): Info C3145: LOOKUP ~regress::single::DG_INDEX found modified record structure
eclcc: /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/ecl/dfse2.ecl(32,12): Info C3146: Field dg_firstname is present in DFS file but not in ECL definition
eclcc: /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/ecl/dfse2.ecl(32,12): Info C3146: Field dg_prange is present in DFS file but not in ECL definition
eclcc: /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/ecl/dfse2.ecl(32,12): Info C3146: Field filepos is present in DFS file but not in ECL definition
eclcc: /home/runner/work/HPCC-Platform/HPCC-Platform/install/opt/HPCCSystems/testing/regress/ecl/dfse2.ecl(33,9): Error C4050: KEYED(dg_lastname) follows unfiltered component dg_firstname in the key '~regress::single::DG_INDEX'
1 error(s), 0 warning(s)
```

@AttilaVamos - it looks like the regression suite didn't spot this and thought still running?

**Comment by AttilaVamos** [comment]
@jakesmith I think Regression Test Engine did spot this dfse2.ecl compiler error here (the result file generated immediately after the archive file generated)  and in all test session before  and after this sesssion. 
It seems something (Regression Test Engine, ecl command, ESP server, etc.) somewhere hanged and only the abort of that workunit/test released it . Unfortunately I can't find any reason from the logs, what and where happened.
I think rerun this smoketest action would finish without any further problem.

**Comment by jakesmith** [comment]
@AttilaVamos - I think we've seen this before?
If it's stuck in esp... can you get the --generateStackTraces - to also capture the esp stacks?

**Comment by AttilaVamos** [comment]
@jakesmith I can.
Strange. From the esp log it seems that workunit appears only when its abort requested.
```
0000173C USR 2021-03-15 12:48:44.329  9352 51138 "Abort W20210315-123636"
0000173D USR 2021-03-15 12:48:44.330  9352 51138 "Abort W20210315-123636 done" 
```

**Comment by jakesmith** [comment]
> @jakesmith I can.
> Strange. From the esp log it seems that workunit appears only when its abort requested.
> 
> ```
> 0000173C USR 2021-03-15 12:48:44.329  9352 51138 "Abort W20210315-123636"
> 0000173D USR 2021-03-15 12:48:44.330  9352 51138 "Abort W20210315-123636 done" 
> ```

"WUDeploy generates" is at the end of the method, so that may add weight to the idea it was stuck in esp for some reason.
Stacks would help clarify that if happens again.

File: dockerfiles/action.yml [hunk 1]
```diff
@@ -16,6 +16,12 @@ inputs:
   build_type:
     description: 'The build type - Debug, RelWithDebInfo, or Release'
     required: false
+  build_label:
+    description: 'The docker version tag'
+    required: false
+  docker_repo:
+    description: 'The name of the docker repositor to pull and push to/from'
```
Comment (richardkchapman) [inline dockerfiles/action.yml:23]: typo repositor

**Review by richardkchapman — CHANGES_REQUESTED** [review]

**Comment by jakesmith** [comment]
@richardkchapman - corrected typo

**Pull request merged**